packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -481,12 +490,103 @@ export class SCWSigner implements Signer {
       standardErrors.rpc.internal(`client not found for chainId ${this.chain.id}`)
     );
 
+    const ownerAccount = owner.account;
+    const ownerPublicKey =
+      ownerAccount.type === 'local' ? ownerAccount.address : ownerAccount.publicKey;
+    let ownerIndex = await getOwnerIndex({
+      address: subAccount.address,
+      publicKey: ownerPublicKey,
+      client,
+    });
+    const globalAccount = this.accounts[1];
+
+    // If the owner index is not found, we need to add the owner from the global account
+    if (!ownerIndex) {
+      const calls: {
+        to: Address;
+        data: Hex;
+        value: Hex;
+      }[] = [];
+
+      const code = await client.getCode({ address: subAccount.address });
+      if (code === '0x') {
+        // If sub account is not deployed and the owner is not in the initCode,
+        // we need to deploy the sub account before adding the owner
+
+        if (!subAccount.factory || !subAccount.factoryData) {
+          throw standardErrors.rpc.internal('no factory or factory data for sub account');
+        }
+
+        calls.push({
+          to: subAccount.factory,
+          data: subAccount.factoryData,
+          value: toHex(0),
+        });
+      }
+
+      if (ownerAccount.type === 'local') {
+        calls.push({
+          to: subAccount.address,
+          data: encodeFunctionData({
+            abi: abi,
+            functionName: 'addOwnerAddress',
+            args: [ownerAccount.address] as const,
+          }),
+          value: toHex(0),
+        });
+      } else {
+        const [x, y] = decodeAbiParameters(
+          [{ type: 'bytes32' }, { type: 'bytes32' }],
+          ownerAccount.publicKey
+        );
+
+        calls.push({
+          to: subAccount.address,
+          data: encodeFunctionData({
+            abi,
+            functionName: 'addOwnerPublicKey',
+            args: [x, y] as const,
+          }),
+          value: toHex(0),
+        });
+      }
+
+      const request: RequestArguments = {
+        method: 'wallet_sendCalls',
+        params: [
+          {
+            version: 1,
+            calls,
+            chainId: toHex(this.chain.id),
+            from: globalAccount,
+          },
+        ],
+      };
+
+      const callId = (await this.sendRequestToPopup(request)) as string;
```

Comment:
this line will get blocked on mobile browser, as there are time limit on popup opening for which any async logic would break

would like to discuss if possible:
1. to modernizing and standardizing the toast for opening popup after async logic, this could unblock insufficient spend permission's case
2. wonder we should still move the construction of add owner call to FE, we will need to come up w a custom solution to pass signer public key to FE